### PR TITLE
Add a `CallbackModeDiscrete` property to `AnimationMixer` to handle the case of blending Continuous and Discrete tracks

### LINKED ISSUE #8512: Add `CallbackModeDiscrete` to `AnimationMixer` and improve the setting method for `UpdateMode` in `AnimationTrackEditor`
### Describe the project you are working on

Godot Animation features stabilization

### Describe the problem or limitation you are having in your project

The blending method for Discrete values is proprietary and is not under user control.

In 3.x, the behavior of AnimationTree forced the Discrete value to be applied first and the Continuous value to be applied later when blending. Therefore, the Continuous value always took precedence in that case.

Since 4.0, when Discrete and Continuous values were mixed, the Discrete value was given priority, but since the problem https://github.com/godotengine/godot/pull/80813#issuecomment-1723651541 occurred, we have decided to force the conversion to the Continuous value when they are mixed so that they look good for many users from a usability point of view. 

Forcing the conversion is done at the time of track cache creation. All animations in the animation list are iterated when creating the track cache, then animations that are not assigned (blended) will also affect the forced conversion, which may confuse the user with a warning if the user unintentionally changes the UpdateMode.

Related: https://github.com/godotengine/godot/issues/83545

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Make users to better understand and control Discrete mode

#### AnimationTrackEditor

Many users probably do not recognize the difference between Discrete mode and Continuous mode + Nearest.

This is because the default UpdateMode created when keying Int is set to Discrete mode, and most often the UpdateMode is not intentionally set to Discrete mode. Probably would be better to set the default to Continuous mode + Nearest and then insert an UpdateMode same with the Reset value if there is.

Also, it would be fine to have a batch conversion function for UpdateMode, but since AnimationTrackEditor lacks a track selection function, I feel that this improvement is needed first.

By the way, the same improvement may be needed for the angle interpolation mode, but this is much more difficult to coexist than Discrete and Continuous, so I feel we have to continue to issue warnings as we do now. There is a coexistence path of separating the applying process without blending, but not for angles.

#### AnimationMixer

Implement CallbackModeDiscrete in the similar way as the Method track.

- enum CallbackModeDiscrete
  - Dominant
    - When blending Continuous and Discrete values, the Discrete value takes precedence
  - Recessive
    - When blending Continuous and Discrete values, the Continuous value takes precedence
    - AnimationPlayer's Default
  - Force Continuous
    - Always treat Discrete values as Continuous + Nearest
    - AnimationTree's Default

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

It is core behavior

### LINKED ISSUE #83545: The animation_mixer gives a false warning about the value track type
### Godot version

v4.2.beta1.official [b1371806a], v4.2.dev6.official [57a6813bb]

### System information

Godot v4.2.beta1 - Windows 10.0.19045 - GLES3 (Compatibility) - NVIDIA GeForce RTX 2060 (NVIDIA; 31.0.15.3667) - Intel(R) Core(TM) i7-10875H CPU @ 2.30GHz (16 Threads)

### Issue description

When I open a scene with an animation_player and click the AnimationPlayer node, Godot gives me a warning about the type of track value:
```
scene/animation/animation_mixer.cpp:884 - AnimationMixer: 'IdleLeft', Value Track: 'Sprite2D:texture' uses a non-numeric type as key value with UpdateMode.UPDATE_CONTINUOUS. This will not be blended correctly, so it is forced to UpdateMode.UPDATE_DISCRETE. (User)
```
Actually, all texture track use `Discrete` update mode and `Nearest` interpolation mode.

The warning message: 

图片

The animation track mentioned in the warning message:

图片

Only 4.2beta1 and 4.2dev6 was affected with this issue, versions before 4.2dev5 cannot reproduction this issue.

### Steps to reproduce

- Open the reproduction project
- Open the `player/player_noanimtree.tscn`.
- Click the `AnimationPlayer` node.
- You can see the warning message as above in the `Output` panel.

### Minimal reproduction project

[min_reproduction.zip](https://github.com/godotengine/godot/files/13000305/min_reproduction.zip)

### PR DESCRIPTION
**Pull request by TokageItLab**
- Closes https://github.com/godotengine/godot-proposals/issues/8512
- Closes https://github.com/godotengine/godot/issues/83545

As noted in https://github.com/godotengine/godot-proposals/issues/8512, this allows user control how the Discrete/Continuous tracks are handled when blending occurs. See also https://github.com/godotengine/godot/pull/86630.

### CODE REVIEW

File: doc/classes/AnimationMixer.xml [hunk 1]
```diff
@@ -273,6 +273,8 @@
 			The number of possible simultaneous sounds for each of the assigned AudioStreamPlayers.
 			For example, if this value is [code]32[/code] and the animation has two audio tracks, the two [AudioStreamPlayer]s assigned can play simultaneously up to [code]32[/code] voices each.
 		</member>
+		<member name="callback_mode_discrete" type="int" setter="set_callback_mode_discrete" getter="get_callback_mode_discrete" enum="AnimationMixer.AnimationCallbackModeDiscrete" default="1">
+		</member>
```
Comment (lyuma) [inline doc/classes/AnimationMixer.xml:277]:
```suggestion
		<member name="callback_mode_discrete" type="int" setter="set_callback_mode_discrete" getter="get_callback_mode_discrete" enum="AnimationMixer.AnimationCallbackModeDiscrete" default="1">
			Ordinarily, tracks can be set to discrete update mode to update infrequently, usually when using nearest interpolation.
			However, to blend between animations correctly, it is recommended to set this to `ANIMATION_CALLBACK_MODE_DISCRETE_FORCE_CONTINUOUS` to update every frame during blending. Other values exist for compatibility but may produce artifacts.
		</member>
```

File: doc/classes/AnimationMixer.xml [hunk 2]
```diff
@@ -350,5 +352,14 @@
 		<constant name="ANIMATION_CALLBACK_MODE_METHOD_IMMEDIATE" value="1" enum="AnimationCallbackModeMethod">
 			Make method calls immediately when reached in the animation.
 		</constant>
+		<constant name="ANIMATION_CALLBACK_MODE_DISCRETE_DOMINANT" value="0" enum="AnimationCallbackModeDiscrete">
+			An [constant Animation.UPDATE_DISCRETE] track value takes precedence when blending the [constant Animation.UPDATE_CONTINUOUS] or [constant Animation.UPDATE_CAPTURE] track values and the [constant Animation.UPDATE_DISCRETE] track values.
+		</constant>
+		<constant name="ANIMATION_CALLBACK_MODE_DISCRETE_RECESSIVE" value="1" enum="AnimationCallbackModeDiscrete">
+			An [constant Animation.UPDATE_CONTINUOUS] or [constant Animation.UPDATE_CAPTURE] track value takes precedence when blending the [constant Animation.UPDATE_CONTINUOUS] or [constant Animation.UPDATE_CAPTURE] track values and the [constant Animation.UPDATE_DISCRETE] track values.
```
Comment (lyuma) [inline doc/classes/AnimationMixer.xml:359]:
```suggestion
			An [constant Animation.UPDATE_CONTINUOUS] or [constant Animation.UPDATE_CAPTURE] track value takes precedence when blending the [constant Animation.UPDATE_CONTINUOUS] or [constant Animation.UPDATE_CAPTURE] track values and the [constant Animation.UPDATE_DISCRETE] track values. This is the default AnimationPlayer behavior in Godot 4.1 and earlier
```

File: doc/classes/AnimationMixer.xml [hunk 3]
```diff
@@ -350,5 +352,14 @@
...
+		</constant>
+		<constant name="ANIMATION_CALLBACK_MODE_DISCRETE_FORCE_CONTINUOUS" value="2" enum="AnimationCallbackModeDiscrete">
+			Always treat the [constant Animation.UPDATE_DISCRETE] track value as [constant Animation.UPDATE_CONTINUOUS] with [constant Animation.INTERPOLATION_NEAREST].
+		</constant>
```
Comment (lyuma) [inline doc/classes/AnimationMixer.xml:363]:
```suggestion
		<constant name="ANIMATION_CALLBACK_MODE_DISCRETE_FORCE_CONTINUOUS" value="2" enum="AnimationCallbackModeDiscrete">
			Always treat the [constant Animation.UPDATE_DISCRETE] track value as [constant Animation.UPDATE_CONTINUOUS] with [constant Animation.INTERPOLATION_NEAREST]. This is the default behavior for AnimationTree.
		</constant>
```

**Review by lyuma — APPROVED** [review]
After Tokage explained it to me, I understand what this is fixing and it makes sense.

I do think users are going to be confused, so I'm suggesting some documentation changes.

File: doc/classes/AnimationMixer.xml (hunk 2 above)
Comment (lyuma) [reply doc/classes/AnimationMixer.xml:359]: This is the default AnimationPlayer behavior in Godot 4.1 and earlier, and in Godot 4.3 and later.

Comment (TokageItLab) [reply doc/classes/AnimationMixer.xml]: References to specific versions may be confusing. So I would like to mention it in a some article separate from class reference.

**Review by AThousandShips — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**